sdk/rust/core/src/vaa/nft.rs

```diff
@@ -0,0 +1,134 @@
+//! This module exposes parsers for NFT bridge VAAs. Token bridging relies on VAA's that indicate
+//! custody/lockup/burn events in order to maintain token parity between multiple chains. These
+//! parsers can be used to read these VAAs. It also defines the Governance actions that this module
+//! supports, namely contract upgrades and chain registrations.
+
+use nom::bytes::complete::take;
+use nom::combinator::verify;
+use nom::number::complete::u8;
+use nom::{
+    Finish,
+    IResult,
+};
+use primitive_types::U256;
+use std::str::from_utf8;
+
+use crate::vaa::{
+    parse_chain,
+    parse_fixed,
+    GovernanceAction,
+};
+use crate::{
+    parse_fixed_utf8,
+    Chain,
+    WormholeError,
+};
+
+/// Transfer is a message containing specifics detailing a token lock up on a sending chain. Chains
+/// that are attempting to initiate a transfer must lock up tokens in some manner, such as in a
+/// custody account or via burning, before emitting this message.
+#[derive(PartialEq, Debug, Clone)]
+pub struct Transfer {
+    /// Address of the token. Left-zero-padded if shorter than 32 bytes
+    pub nft_address: [u8; 32],
+
+    /// Chain ID of the token
+    pub nft_chain: Chain,
+
+    /// Symbol of the token
+    pub symbol: String,
```

Comment:
Aren't `symbol` and `name` supposed to be [u8; 32]? The nft spec says so

Reply:
On the receiving end, `String` is nicer for sure, but when you construct one of these, I think it's better if the type tells you about the size limitation explicitly as it's an important detail in the spec